### PR TITLE
V2: Upgrade to Next.js v15

### PR DESCRIPTION
**Pull request by timostamm**
Upgrading Next.js following the [docs](https://nextjs.org/docs/canary/app/building-your-application/upgrading/version-15) with `npx @next/codemod@canary upgrade latest`.

I don't see any issues. There's one notable improvement in RSC: Next.js v15 correctly hydrates and serializes Uint8Array when crossing the boundary to a client component.

Note that this PR will fail install in CI because of the peer dependency constraint in `@connectrpc/connect-next`: `"next": "^13.2.4 || ^14.2.5"`. After https://github.com/connectrpc/connect-es/pull/1290 is merged and release, we need to update the dependencies in this PR, which should turn it green.

### CODE REVIEW

File: nextjs/pages/ssr.tsx [hunk 1]
```diff
@@ -36,7 +36,7 @@ export const getServerSideProps = async () => {
       response,
 
       // The message `Payload` uses values that Next.js cannot serialize to JSON -
-      // bigint, Infinity, and typed arrays. proto2 messages use the prototype
+      // BigInt, Infinity, and Uint8Array. proto2 messages use the prototype
```
Comment (smaye81) [inline nextjs/pages/ssr.tsx:39]: I'm probably misunderstanding either how Next.js works or how we have our examples set up, but I think this sentence conflicts with the new Next.js v15 features and the comment in `nextjs/app/boundary/page.tsx`

**Review by smaye81 — APPROVED** [review]

File: nextjs/pages/ssr.tsx (hunk 1 above)
Comment (timostamm) [reply nextjs/pages/ssr.tsx:39]: getServerSideProps and RSC are two completely separate things. 

Details: https://github.com/connectrpc/connectrpc.com/pull/218 -> preview -> docs -> web -> SSR.

Comment (smaye81) [reply nextjs/pages/ssr.tsx:39]: That's what I figured. 👍

**Comment by timostamm** [comment]
Updated dependencies. CI is green.

**Pull request merged**